src/main/java/collab/CommandParser.java

```diff
@@ -1,37 +1,113 @@
+
+    protected static final List<String> WHITE_COMMAND_LIST = Arrays.asList("ADD", "SCH", "MOD", "DEL");
+    protected static final List<String> FIRST_OPTION_WHITE_LIST = Arrays.asList(" ", "-p");
+    protected static final Map<String, List<String>> SECOND_OPTION_WHITE_MAP = new HashMap<>(){{
+        put("-f", Arrays.asList("name"));
+        put("-l", Arrays.asList("name", "phoneNum"));
+        put("-m", Arrays.asList("phoneNum", "birthday"));
+        put("-y", Arrays.asList("birthday"));
+        put("-d", Arrays.asList("birthday"));
+    }};
+
+    protected static final int ADD_CMD_LENGTH = 10;
+    protected static final int SCH_CMD_LENGTH = 6;
+    protected static final int MOD_CMD_LENGTH = 8;
+    protected static final int DEL_CMD_LENGTH = 6;
+
+    protected static final String ERR_MSG_CMD_INVALID = "ERROR::Command is invalid...";
+    protected static final String ERR_MSG_FORMAT_WRONG = "ERROR::Command format is wrong...";
+    protected static final String ERR_MSG_OPTION_WRONG = "ERROR::Command option is wrong...";
+
+
+
+    public List<ICommand>  parseToCommandList(List<String> cmdStrList) throws Exception{
+        List<ICommand> commandList = new ArrayList<ICommand>();
         for (String cmdStrLine : cmdStrList){
-            String[] tokenList = cmdStrLine.split(",");
-            //getCommandFromToken(getFirstOptionFromToken(tokenList), getSecondOptionFromToken(tokenList), getThirdOptionFromToken(tokenList), tokenList);
+            commandList.add(parseToCommand(cmdStrLine));
         }
-        return null;
+        return commandList;
     }
-    public ICommand  parseToCommand(String cmdStr){
+    public ICommand  parseToCommand(String cmdStr) throws Exception {
         String[] tokenList = cmdStr.split(",");
         checkValidation(tokenList);
-        //return getCommandFromToken(getFirstOptionFromToken(tokenList), getSecondOptionFromToken(tokenList), getThirdOptionFromToken(tokenList), tokenList);
-        return new AddCommand(null, null, null, null);
+        return getCommandFromToken(getFirstOptionFromToken(tokenList), getSecondOptionFromToken(tokenList), getThirdOptionFromToken(tokenList), tokenList);
+    }
+    private void checkValidation(String[] tokenList) throws Exception {
+        if (tokenList.length < 1) throw new Exception(ERR_MSG_FORMAT_WRONG);
+        if (!WHITE_COMMAND_LIST.contains(tokenList[0])) throw new Exception(ERR_MSG_CMD_INVALID);
+        if (!FIRST_OPTION_WHITE_LIST.contains(tokenList[1])) throw new Exception(ERR_MSG_OPTION_WRONG);
+        if (!tokenList[2].equals(" ")) checkSecondOptionValidation(tokenList);
+        if (tokenList[0].equals("ADD")) checkAddCommandValidation(tokenList);
+        if (tokenList[0].equals("SCH")) checkSearchCommandValidation(tokenList);
+        if (tokenList[0].equals("MOD")) checkModifyCommandValidation(tokenList);
+        if (tokenList[0].equals("DEL")) checkDeleteCommandValidation(tokenList);
```

Comment:
향후 명령어의 단어가 변경될 경우를 고려해서 이 부분을 문자열보다는 값을 읽어오게 하는 것은 어떨까요?
SCH를 FND로 바꾸라는 요구사항이 만약 있다면 수정하기 편하지 않을까 생각합니다.
한편으로는 기존 코드가 명확하게 무슨 명령어인지 바로 표시 해 주니, 가독성은 기존이 좀 더 좋을 것 같아요.


```suggestion
        if (tokenList[0].equals(WHITE_COMMAND_LIST.get(0))) checkAddCommandValidation(tokenList);
        if (tokenList[0].equals(WHITE_COMMAND_LIST.get(1))) checkSearchCommandValidation(tokenList);
        if (tokenList[0].equals(WHITE_COMMAND_LIST.get(2))) checkModifyCommandValidation(tokenList);
        if (tokenList[0].equals(WHITE_COMMAND_LIST.get(3))) checkDeleteCommandValidation(tokenList);
```

Reply:
좋은 의견 감사 드립니다. 수정하여 반영해 두도록 하겠습니다~